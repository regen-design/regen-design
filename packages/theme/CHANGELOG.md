# @regen-design/theme

## 1.0.0

### Major Changes

- publish
