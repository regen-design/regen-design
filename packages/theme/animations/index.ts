import { keyframes } from 'styled-components'

export const ButtonWaveSpreadAnimationKeyframes = (color: string) => keyframes`
    0% {
        opacity: 0.6;
        box-shadow: 0 0 1px 0 ${color};
    }
    100% {
        opacity: 0;
        box-shadow: 0 0 1px 5px ${color};
    }
`
