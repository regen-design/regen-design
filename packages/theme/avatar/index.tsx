import { NAME_SPACE } from '@regen-design/constant'
import { AvatarProps } from '@regen-design/types'
import styled from 'styled-components'
import { convertTheme } from '../tools'
const prefix = `${NAME_SPACE}-avatar`
export const StyledAvatarPrefixClass = prefix
export const StyledAvatar = styled.div<AvatarProps>`
  display: inline-flex;
  position: relative;
  overflow: hidden;
  text-align: center;
  border: ${props => {
    const _theme = convertTheme(props.theme)
    return `1px solid ${_theme.colors[props.borderColor || 'transparent']}`
  }};
  width: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.components.avatar.size[props.size || 'default']
  }};
  height: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.components.avatar.size[props.size || 'default']
  }};
  color: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.colors[props.color || 'white']
  }};
  font-size: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.components.avatar.fontSize
  }};
  border-radius: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.rounded) return '50%'
    return _theme.borderRadius
  }};
  background-color: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.colors[props.backgroundColor || 'placeholder']
  }};
  img {
    width: 100%;
    height: 100%;
  }
  .${prefix}__text {
    position: absolute;
    line-height: 1;
    white-space: nowrap;
    display: inline-block;
    left: 50%;
    top: 50%;
  }
`
