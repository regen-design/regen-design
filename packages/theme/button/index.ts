import styled from 'styled-components'
import { convertTheme } from '../tools'
import { ButtonProps } from '@regen-design/types'
import { NAME_SPACE } from '@regen-design/constant'
import { lighten } from 'polished'
const prefix = `${NAME_SPACE}-button`

export const StyledButtonPrefixClass = prefix

export const StyledButton = styled.button<ButtonProps>`
  background-color: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.text) {
      return 'transparent'
    }
    if (props.type === 'default') {
      return _theme.theme === 'light' ? _theme.colors.light : _theme.colors.dark
    }
    if (props.dashed) {
      return 'transparent'
    }
    return _theme.colors[props.type]
  }};
  border-radius: ${props => {
    return convertTheme(props.theme).borderRadius + 'px'
  }};
  color: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.text) {
      return _theme.colors[props.type]
    }
    if (props.type === 'default') {
      return _theme.theme === 'light' ? _theme.colors.dark : _theme.colors.light
    }
    if (props.dashed) {
      return `${_theme.colors[props.type]}`
    }
    return _theme.theme === 'light' ? _theme.colors.light : _theme.colors.dark
  }};
  &:hover {
    color: ${props => {
      const _theme = convertTheme(props.theme)
      if (props.type === 'default') {
        return lighten(0.025, _theme.colors.primary)
      }
      if (props.text) {
        return lighten(0.05, _theme.colors?.[props.type] || '#000000')
      }
    }};
  }
  &:active {
    color: ${props => {
      const _theme = convertTheme(props.theme)
      if (props.type === 'default') {
        return lighten(0.15, _theme.colors.primary)
      }
      if (props.text) {
        return lighten(0.25, _theme.colors?.[props.type] || '#000000')
      }
      if (props.dashed) {
        return `${_theme.colors[props.type]}`
      }
      return _theme.theme === 'light' ? _theme.colors.light : _theme.colors.dark
    }};
  }
  transition: ${props => {
    const duration = `${convertTheme(props.theme).animations.duration}ms`
    return `all ${props.text ? '50ms' : duration} ease-in`
  }};
  opacity: ${props => {
    if (props.disabled) {
      return 0.5
    }
  }};
  display: inline-flex;
  flex-wrap: nowrap;
  white-space: nowrap;
  outline: none;
  position: relative;
  z-index: auto;
  border: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.text) {
      return 'transparent'
    }
    if (props.type === 'default') {
      const color = _theme.theme === 'light' ? _theme.borderColor : _theme.borderDarkColor
      return `1px ${props.dashed ? 'dashed' : 'solid'} ${color}`
    }
    if (props.dashed) {
      return `1px dashed ${_theme.colors[props.type]}`
    }
    return '1px solid transparent'
  }};
  padding: ${props => {
    if (props.text) {
      return '0 0'
    }
    switch (props.size) {
      case 'tiny':
        return '0 8px'
      case 'small':
        return '0 12px'
      case 'large':
        return '0 16px'
      default:
        return '0 12px'
    }
  }};
  height: ${props => {
    if (props.text) {
      return 'initial'
    }
    switch (props.size) {
      case 'tiny':
        return '24px'
      case 'small':
        return '32px'
      case 'large':
        return '40px'
      default:
        return '36px'
    }
  }};
  line-height: 1;
  font-family: inherit;
  font-size: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.fontSizes[props.size]
  }};
  width: initial;
  align-items: center;
  justify-content: center;
  user-select: none;
  -webkit-user-select: none;
  text-align: center;
  cursor: ${props => {
    if (props.disabled) {
      return 'not-allowed'
    }
    return 'pointer'
  }};
  text-decoration: none;
  .${prefix}__content {
    display: flex;
    align-items: center;
    flex-wrap: nowrap;
    min-width: 0;
  }
  .${prefix}__icon {
    margin-right: 8px;
  }
`
