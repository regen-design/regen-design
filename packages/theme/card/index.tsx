import { NAME_SPACE } from '@regen-design/constant'
import { CardProps } from '@regen-design/types'
import styled from 'styled-components'
import { convertTheme } from '../tools'
const prefix = `${NAME_SPACE}-card`
export const StyledCardPrefixClass = prefix
export const StyledCard = styled.div<CardProps>`
  border: ${props => {
    const _theme = convertTheme(props.theme)
    return `1px solid ${_theme.theme === 'light' ? _theme.borderColor : _theme.borderDarkColor}`
  }};
  border-radius: ${props => convertTheme(props.theme).borderRadius};
  border-bottom: 0;
  width: 100%;
  box-sizing: border-box;
  position: relative;
  > .${prefix}-header {
    padding: ${props => {
      const _theme = convertTheme(props.theme)
      return `${_theme.paddingSizes[props.size]}`
    }};
    display: flex;
    justify-content: space-between;
    align-items: center;
    > .${prefix}-title {
      font-size: ${props => {
        const _theme = convertTheme(props.theme)
        return `${_theme.fontSizes[props.size]}`
      }};
      color: ${props => props.theme.textColor};
    }
  }

  > .${prefix}-body {
    > .${prefix}-description {
      margin-bottom: ${props => {
        const _theme = convertTheme(props.theme)
        return `${_theme.paddingSizes[props.size]}`
      }};
    }
    padding: ${props => {
      const _theme = convertTheme(props.theme)
      const size = _theme.paddingSizes[props.size]
      return `0 ${size} ${size}`
    }};
    border-bottom: ${props => {
      const _theme = convertTheme(props.theme)
      return `1px solid ${_theme.theme === 'light' ? _theme.borderColor : _theme.borderDarkColor}`
    }};
  }
  > .${prefix}-footer {
    position: relative;
    border-bottom: ${props => {
      const _theme = convertTheme(props.theme)
      const color = _theme.theme === 'dark' ? _theme.borderDarkColor : _theme.borderColor
      return `1px solid ${color}`
    }};
    padding: ${props => {
      const _theme = convertTheme(props.theme)
      const size = _theme.paddingSizes[props.size]
      return `${size}`
    }};
  }
`
