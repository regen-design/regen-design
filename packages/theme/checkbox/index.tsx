import { NAME_SPACE } from '@regen-design/constant'
import { CheckboxProps } from '@regen-design/types'
import styled from 'styled-components'
import { convertTheme } from '../tools'
import { lighten, rgba } from 'polished'

const prefix = `${NAME_SPACE}-checkbox`
export const StyledCheckboxPrefixClass = prefix
export const StyledCheckbox = styled.div<CheckboxProps>`
  display: inline-flex;
  flex-wrap: nowrap;
  word-break: break-word;
  align-items: flex-start;
  line-height: 1.5;
  font-size: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.fontSizes[props.size]
  }};
  cursor: ${props => {
    if (props.disabled) {
      return 'not-allowed'
    }
    return 'pointer'
  }};
  margin: 0;
  padding: 0;
  &:not(.${prefix}--disabled):focus {
    .${prefix}-wrapper {
      .${prefix}-box__border {
        border: ${props => {
          const _theme = convertTheme(props.theme)
          return `1px solid ${_theme.colors['primary']}`
        }};
      }
      .${prefix}-box {
        box-shadow: ${props => {
          const _theme = convertTheme(props.theme)
          const color = _theme.colors.primary
          return `0 0 0 ${_theme.waveBlurRadius / 2}px ${rgba(color, 0.15)}`
        }};
      }
    }
  }
  .${prefix}-label {
    user-select: none;
    -webkit-user-select: none;
    padding: 0 4px;
    line-height: 1.5;
    color: ${props => {
      const _theme = convertTheme(props.theme)
      if (props.disabled) {
        return _theme.colors.placeholder
      }
      return 'inherit'
    }};
  }
  .${prefix}-wrapper {
    .${prefix}-box {
      background-color: ${props => {
        const _theme = convertTheme(props.theme)
        if (props.disabled) {
          if (_theme.theme === 'dark') {
            return rgba('#ffffff', 0.1)
          }
          return _theme.colors.light
        }
        if (props.indeterminate || props.checked) {
          return _theme.colors.primary
        }
        return 'inherit'
      }};
      transition:
        background-color 0.3s ease,
        box-shadow 0.3s ease;
      border-radius: inherit;
      width: ${props => {
        const _theme = convertTheme(props.theme)
        return _theme.fontSizes[props.size]
      }};
      height: ${props => {
        const _theme = convertTheme(props.theme)
        return _theme.fontSizes[props.size]
      }};
      position: absolute;
      left: 0;
      top: 50%;
      transform: translateY(-50%);
    }
    .${prefix}-icon {
      color: ${props => {
        const _theme = convertTheme(props.theme)
        if (props.disabled) {
          if (_theme.theme === 'dark') {
            return lighten(0.25, _theme.colors.dark)
          }
          return _theme.colors.placeholder
        }
        return _theme.colors.light
      }};
      transform-origin: center center;
      display: flex;
      align-items: center;
      justify-content: center;
      position: absolute;
      width: 100%;
      height: 100%;
      padding: 2px;
      font-weight: bold;
      font-size: ${props => {
        const _theme = convertTheme(props.theme)
        return `calc(${_theme.fontSizes[props.size]} )`
      }};
    }
    .${prefix}-box__border {
      width: 100%;
      height: 100%;
      border-radius: inherit;
      border: 1px solid
        ${props => {
          const _theme = convertTheme(props.theme)
          if (_theme.theme === 'dark') {
            return _theme.borderDarkColor
          }
          if (props.disabled) {
            return _theme.colors.placeholder
          }
          if (props.indeterminate || props.checked) {
            return _theme.colors.primary
          }
          return _theme.borderColor
        }};
      transition: border-color 0.3s ease;
    }
    box-sizing: border-box;
    position: relative;
    flex-shrink: 0;
    flex-grow: 0;
    user-select: none;
    width: ${props => {
      const _theme = convertTheme(props.theme)
      return _theme.fontSizes[props.size]
    }};
    -webkit-user-select: none;
    white-space: nowrap;
    display: inline-block;
    line-height: 1.5;
    border-radius: ${props => {
      const _theme = convertTheme(props.theme)
      return _theme.borderRadius
    }};
  }
`
