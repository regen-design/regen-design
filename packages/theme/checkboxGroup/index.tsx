import { NAME_SPACE } from '@regen-design/constant'
import { CheckboxGroupProps } from '@regen-design/types'
import styled from 'styled-components'
const prefix = `${NAME_SPACE}-checkboxGroup`
export const StyledCheckboxGroupPrefixClass = prefix
export const StyledCheckboxGroup = styled.div<CheckboxGroupProps>``
