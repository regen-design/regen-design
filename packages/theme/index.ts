import { ThemeConfig } from '@regen-design/types'
import { useTheme as useThemeFn } from 'styled-components'
export const useTheme = useThemeFn as () => ThemeConfig
export * from './animations'
export * from './button'
export * from './card'
export * from './checkbox'
export * from './checkboxGroup'
export * from './codeBlock'
export * from './grid'
export * from './gridItem'
export * from './icon'
export * from './input'
export * from './radio'
export * from './radioGroup'
export * from './space'
export * from './variable'
export * from './wave'
