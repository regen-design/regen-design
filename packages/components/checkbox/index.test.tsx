import '@testing-library/jest-dom/vitest'
import { describe, test, expect, vitest } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/react'
import { Checkbox } from './index'
import { StyledCheckboxPrefixClass } from '@regen-design/theme'
describe('Checkbox', () => {
  test('should render correctly', () => {
    const element = render(<Checkbox />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('checkbox')).toBeInTheDocument()
  })
  test('should render checked state correctly', () => {
    const element = render(<Checkbox checked />)
    expect(element).toMatchSnapshot()
    expect(
      screen.getByRole('checkbox').querySelector("svg[data-icon*='check']")
    ).toBeInTheDocument()
  })
  test('should render indeterminate state correctly', () => {
    const element = render(<Checkbox indeterminate />)
    expect(element).toMatchSnapshot()
    expect(
      screen.getByRole('checkbox').querySelector("svg[data-icon*='minus']")
    ).toBeInTheDocument()
  })
  test('should render disabled state correctly', () => {
    const onClickMock = vitest.fn()
    const element = render(<Checkbox disabled />)
    expect(element).toMatchSnapshot()
    fireEvent.click(element.getByRole('checkbox'))
    expect(onClickMock).not.toHaveBeenCalled()
  })
  test('should trigger onChange event', () => {
    let checked = false
    const onChangeMock = vitest.fn(value => (checked = value))
    const element = render(<Checkbox checked={checked} onChange={onChangeMock} />)
    expect(element).toMatchSnapshot()
    fireEvent.click(element.getByRole('checkbox'))
    expect(checked).toBe(true)
  })
  test('should render custom size correctly', () => {
    const element = render(<Checkbox size="small" />)
    expect(element).toMatchSnapshot()
    expect(screen.getByRole('checkbox')).toHaveClass(`${StyledCheckboxPrefixClass}--small`)
  })
})
