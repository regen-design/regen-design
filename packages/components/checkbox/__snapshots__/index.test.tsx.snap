// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox > should render checked state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eDLKkx bXwzQb regen-checkbox"
        role="checkbox"
        tabindex="0"
      >
        <div
          class="regen-checkbox-wrapper"
        >
           
          <div
            class="regen-checkbox-box"
          >
            <div
              aria-hidden="true"
              class="sc-jIBlqr khsxkQ regen-wave "
              role="wave"
              type="default"
            />
            <div
              class="regen-checkbox-icon"
              style="transform: scale(0) translateZ(0);"
            >
              <i
                class="sc-fmKFGs fekwRS regen-icon regen-icon-check"
                color="currentColor"
                role="icon"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check"
                  style="width: 1em; height: 1em;"
                  viewBox="0 0 507.506 507.506"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M163.865,436.934c-14.406,0.006-28.222-5.72-38.4-15.915L9.369,304.966c-12.492-12.496-12.492-32.752,0-45.248l0,0   c12.496-12.492,32.752-12.492,45.248,0l109.248,109.248L452.889,79.942c12.496-12.492,32.752-12.492,45.248,0l0,0   c12.492,12.496,12.492,32.752,0,45.248L202.265,421.019C192.087,431.214,178.271,436.94,163.865,436.934z"
                    />
                  </g>
                </svg>
              </i>
            </div>
            <div
              class="regen-checkbox-box__border"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eDLKkx bXwzQb regen-checkbox"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="regen-checkbox-wrapper"
      >
         
        <div
          class="regen-checkbox-box"
        >
          <div
            aria-hidden="true"
            class="sc-jIBlqr khsxkQ regen-wave "
            role="wave"
            type="default"
          />
          <div
            class="regen-checkbox-icon"
            style="transform: scale(0) translateZ(0);"
          >
            <i
              class="sc-fmKFGs fekwRS regen-icon regen-icon-check"
              color="currentColor"
              role="icon"
            >
              <svg
                aria-hidden="true"
                data-icon="check"
                style="width: 1em; height: 1em;"
                viewBox="0 0 507.506 507.506"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M163.865,436.934c-14.406,0.006-28.222-5.72-38.4-15.915L9.369,304.966c-12.492-12.496-12.492-32.752,0-45.248l0,0   c12.496-12.492,32.752-12.492,45.248,0l109.248,109.248L452.889,79.942c12.496-12.492,32.752-12.492,45.248,0l0,0   c12.492,12.496,12.492,32.752,0,45.248L202.265,421.019C192.087,431.214,178.271,436.94,163.865,436.934z"
                  />
                </g>
              </svg>
            </i>
          </div>
          <div
            class="regen-checkbox-box__border"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox > should render custom size correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eDLKkx kOktTj regen-checkbox regen-checkbox--small"
        role="checkbox"
        tabindex="0"
      >
        <div
          class="regen-checkbox-wrapper"
        >
           
          <div
            class="regen-checkbox-box"
          >
            <div
              aria-hidden="true"
              class="sc-jIBlqr khsxkQ regen-wave "
              role="wave"
              type="default"
            />
            <div
              class="regen-checkbox-icon"
              style="transform: scale(0) translateZ(0);"
            />
            <div
              class="regen-checkbox-box__border"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eDLKkx kOktTj regen-checkbox regen-checkbox--small"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="regen-checkbox-wrapper"
      >
         
        <div
          class="regen-checkbox-box"
        >
          <div
            aria-hidden="true"
            class="sc-jIBlqr khsxkQ regen-wave "
            role="wave"
            type="default"
          />
          <div
            class="regen-checkbox-icon"
            style="transform: scale(0) translateZ(0);"
          />
          <div
            class="regen-checkbox-box__border"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox > should render disabled state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eDLKkx iRVmvr regen-checkbox regen-checkbox--disabled"
        disabled=""
        role="checkbox"
        tabindex="0"
      >
        <div
          class="regen-checkbox-wrapper"
        >
           
          <div
            class="regen-checkbox-box"
          >
            <div
              aria-hidden="true"
              class="sc-jIBlqr khsxkQ regen-wave "
              role="wave"
              type="default"
            />
            <div
              class="regen-checkbox-icon"
              style="transform: scale(0) translateZ(0);"
            />
            <div
              class="regen-checkbox-box__border"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eDLKkx iRVmvr regen-checkbox regen-checkbox--disabled"
      disabled=""
      role="checkbox"
      tabindex="0"
    >
      <div
        class="regen-checkbox-wrapper"
      >
         
        <div
          class="regen-checkbox-box"
        >
          <div
            aria-hidden="true"
            class="sc-jIBlqr khsxkQ regen-wave "
            role="wave"
            type="default"
          />
          <div
            class="regen-checkbox-icon"
            style="transform: scale(0) translateZ(0);"
          />
          <div
            class="regen-checkbox-box__border"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox > should render indeterminate state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eDLKkx bXwzQb regen-checkbox regen-checkbox--indeterminate"
        role="checkbox"
        tabindex="0"
      >
        <div
          class="regen-checkbox-wrapper"
        >
           
          <div
            class="regen-checkbox-box"
          >
            <div
              aria-hidden="true"
              class="sc-jIBlqr khsxkQ regen-wave "
              role="wave"
              type="default"
            />
            <div
              class="regen-checkbox-icon"
              style="transform: scale(0) translateZ(0);"
            >
              <i
                class="sc-fmKFGs fekwRS regen-icon regen-icon-minus"
                color="currentColor"
                role="icon"
              >
                <svg
                  aria-hidden="true"
                  data-icon="minus"
                  style="width: 1em; height: 1em;"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="2"
                    rx="1"
                    width="24"
                    y="11"
                  />
                </svg>
              </i>
            </div>
            <div
              class="regen-checkbox-box__border"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eDLKkx bXwzQb regen-checkbox regen-checkbox--indeterminate"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="regen-checkbox-wrapper"
      >
         
        <div
          class="regen-checkbox-box"
        >
          <div
            aria-hidden="true"
            class="sc-jIBlqr khsxkQ regen-wave "
            role="wave"
            type="default"
          />
          <div
            class="regen-checkbox-icon"
            style="transform: scale(0) translateZ(0);"
          >
            <i
              class="sc-fmKFGs fekwRS regen-icon regen-icon-minus"
              color="currentColor"
              role="icon"
            >
              <svg
                aria-hidden="true"
                data-icon="minus"
                style="width: 1em; height: 1em;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="2"
                  rx="1"
                  width="24"
                  y="11"
                />
              </svg>
            </i>
          </div>
          <div
            class="regen-checkbox-box__border"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox > should trigger onChange event 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eDLKkx kOktTj regen-checkbox"
        role="checkbox"
        tabindex="0"
      >
        <div
          class="regen-checkbox-wrapper"
        >
           
          <div
            class="regen-checkbox-box"
          >
            <div
              aria-hidden="true"
              class="sc-jIBlqr khsxkQ regen-wave "
              role="wave"
              type="default"
            />
            <div
              class="regen-checkbox-icon"
              style="transform: scale(0) translateZ(0);"
            />
            <div
              class="regen-checkbox-box__border"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eDLKkx kOktTj regen-checkbox"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="regen-checkbox-wrapper"
      >
         
        <div
          class="regen-checkbox-box"
        >
          <div
            aria-hidden="true"
            class="sc-jIBlqr khsxkQ regen-wave "
            role="wave"
            type="default"
          />
          <div
            class="regen-checkbox-icon"
            style="transform: scale(0) translateZ(0);"
          />
          <div
            class="regen-checkbox-box__border"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
